language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/FieldAccessCompletionResolver.java

```diff
@@ -121,6 +123,9 @@ public FieldAccessCompletionResolver(PositionedOperationContext context, boolean
         Predicate<Symbol> predicate = symbol -> symbol.kind() == SymbolKind.METHOD
                 || symbol.kind() == SymbolKind.FUNCTION;
         String methodName = ((SimpleNameReferenceNode) nameRef).name().text();
+        if (methodName.startsWith(IDENTIFIER_LITERAL_PREFIX)) {
+            methodName = IdentifierUtils.unescapeUnicodeCodepoints(methodName.substring(1));
+        }
```

Comment:
Instead of special casing this in one place, shall we follow a similar approach to the node transformer and fix all the places in this visitor? i.e., Move this name extraction logic to a method and use that method anywhere a name is accessed in this visitor

Reply:
@pubudu91 But, usually we don't change the attributes of the nodes in the Syntax tree, which we get from the parser. Right? Here, In the `FieldAccessCompletionResolver.java`, we get the method name from a `MethodCallExpressionNode`. :thinking:

Reply:
We aren't changing anything. But since the compiler keeps all identifiers in unescaped form, need to use the unescaped form when comparing the names we get from the syntax tree nodes with names taken from symbols. Basically the idea is, each place in the `FieldAccessCompletionResolver` where we take a name from a syntax tree node, we unescape it first. This fix only unescapes method call expressions.

Reply:
Now I got the idea. Thanks @pubudu91